From mboxrd@z Thu Jan  1 00:00:00 1970
Return-Path: <linux-kernel-owner@kernel.org>
X-Spam-Checker-Version: SpamAssassin 3.4.0 (2014-02-07) on
	aws-us-west-2-korg-lkml-1.web.codeaurora.org
Received: from mail.kernel.org (mail.kernel.org [198.145.29.99])
	by smtp.lore.kernel.org (Postfix) with ESMTP id F233EC433EF
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:50:06 +0000 (UTC)
Received: from vger.kernel.org (vger.kernel.org [23.128.96.18])
	by mail.kernel.org (Postfix) with ESMTP id DAF61611F0
	for <linux-kernel@archiver.kernel.org>; Tue,  9 Nov 2021 09:50:06 +0000 (UTC)
Received: (majordomo@vger.kernel.org) by vger.kernel.org via listexpand
        id S245017AbhKIJwv (ORCPT <rfc822;linux-kernel@archiver.kernel.org>);
        Tue, 9 Nov 2021 04:52:51 -0500
Received: from lindbergh.monkeyblade.net ([23.128.96.19]:49212 "EHLO
        lindbergh.monkeyblade.net" rhost-flags-OK-OK-OK-OK) by vger.kernel.org
        with ESMTP id S244190AbhKIJv1 (ORCPT
        <rfc822;linux-kernel@vger.kernel.org>);
        Tue, 9 Nov 2021 04:51:27 -0500
Received: from mail-pl1-x62e.google.com (mail-pl1-x62e.google.com [IPv6:2607:f8b0:4864:20::62e])
        by lindbergh.monkeyblade.net (Postfix) with ESMTPS id D8D94C0613F5
        for <linux-kernel@vger.kernel.org>; Tue,  9 Nov 2021 01:48:38 -0800 (PST)
Received: by mail-pl1-x62e.google.com with SMTP id y7so2881437plp.0
        for <linux-kernel@vger.kernel.org>; Tue, 09 Nov 2021 01:48:38 -0800 (PST)
DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=sifive.com; s=google;
        h=from:to:subject:date:message-id:in-reply-to:references:mime-version
         :content-transfer-encoding;
        bh=PvZSczC6umJopWKeatvFKRPXXAs3l7Ppj7hlaQG6eAc=;
        b=LQcCH0fzxq5GO9KWLodqCj4YR97Vh05wdeUqjoAQnknd1XkF7dILCo6Xo2qYowiqSO
         gmAhRZQJWXrlTVcLPpjnVWwt4TMHUr5ldhEMBgq1p49pCP0O8chBDBmoOMhc83ProboG
         pIuExYUJRn55PEnFv7Gme3NYaTOa9GMM9/gF9K58CHA2CHdTsfHrXi7uZjRWnuGYCP86
         7pOKMpX840K4AjXMeFT8gspwu+FFPmDPllRgfUQkdmz6PRx8jm1ol7m+IiUZ3+xUUqKh
         2nOqiVSwagP+8zfTApZAJ56jokeFgwRZdWtj8JxNljtWftoRNYtgtgVRUdkekqaqNKVz
         aJig==
X-Google-DKIM-Signature: v=1; a=rsa-sha256; c=relaxed/relaxed;
        d=1e100.net; s=20210112;
        h=x-gm-message-state:from:to:subject:date:message-id:in-reply-to
         :references:mime-version:content-transfer-encoding;
        bh=PvZSczC6umJopWKeatvFKRPXXAs3l7Ppj7hlaQG6eAc=;
        b=vsJ/in4KO5H5AnuBGERz18NUEgOxJ4hhoZSpkNYk0HxlJXHeQVQgdvwGFlslBQfagZ
         SQ9/OFS9gZJXOh0U9aftz5PiH/BLuHpSgUOtKBAbuk9uyj0TrcldnGko6kN/XNcbnevX
         /e3MHgV8krca+Yq+kIwliRtESqGk2Eww/LnrZL32RmiyLmxG1dXKcUEOu+0ExgdRkCbf
         KhvLr2Nvt/gQRJAFS2dYK/qoqxnJYdkPCmNRnYaaz6ghHJR5/IlDqzH3i+9IriMieABY
         ltcMrcBVCZubivsl/LBVGDCqSi8nn8DvR1jMNh7cR09hrwsuTF3tt5GOQPSmOFiXZgEp
         PVGQ==
X-Gm-Message-State: AOAM530l/s08r2gNEUuyLGCtpBL9+VyLJ21F+VZpyZxYWpGGs3GPkz79
        JJvBARbE9IH+PyUbTNJJM+UFZQ==
X-Google-Smtp-Source: ABdhPJyoTIXbZWcpshBd5z+5mEStDbBaicx0t7HFnmYLXGaSSThFwCDNKe9kjbVZ7S3dRqv2vIjP0w==
X-Received: by 2002:a17:90b:180b:: with SMTP id lw11mr5896661pjb.108.1636451318438;
        Tue, 09 Nov 2021 01:48:38 -0800 (PST)
Received: from hsinchu16.internal.sifive.com (59-124-168-89.hinet-ip.hinet.net. [59.124.168.89])
        by smtp.gmail.com with ESMTPSA id cv1sm1940511pjb.48.2021.11.09.01.48.37
        (version=TLS1_3 cipher=TLS_AES_256_GCM_SHA384 bits=256/256);
        Tue, 09 Nov 2021 01:48:38 -0800 (PST)
From:   Greentime Hu <greentime.hu@sifive.com>
To:     palmer@dabbelt.com, paul.walmsley@sifive.com,
        linux-riscv@lists.infradead.org, linux-kernel@vger.kernel.org,
        aou@eecs.berkeley.edu
Subject: [PATCH v9 04/17] riscv: Add new csr defines related to vector extension
Date:   Tue,  9 Nov 2021 17:48:16 +0800
Message-Id: <6013bd0c309cecf244a151b54ee73c0ebd391663.1636362169.git.greentime.hu@sifive.com>
X-Mailer: git-send-email 2.31.1
In-Reply-To: <cover.1636362169.git.greentime.hu@sifive.com>
References: <cover.1636362169.git.greentime.hu@sifive.com>
MIME-Version: 1.0
Content-Transfer-Encoding: 8bit
Precedence: bulk
List-ID: <linux-kernel.vger.kernel.org>
X-Mailing-List: linux-kernel@vger.kernel.org

Follow the riscv vector spec to add new csr numbers.

[guoren@linux.alibaba.com: first porting for new vector related csr]
Signed-off-by: Greentime Hu <greentime.hu@sifive.com>
Acked-by: Guo Ren <guoren@kernel.org>
Co-developed-by: Guo Ren <guoren@linux.alibaba.com>
Signed-off-by: Guo Ren <guoren@linux.alibaba.com>
Co-developed-by: Vincent Chen <vincent.chen@sifive.com>
Signed-off-by: Vincent Chen <vincent.chen@sifive.com>
---
 arch/riscv/include/asm/csr.h | 16 ++++++++++++++--
 1 file changed, 14 insertions(+), 2 deletions(-)

diff --git a/arch/riscv/include/asm/csr.h b/arch/riscv/include/asm/csr.h
index 87ac65696871..069743102fac 100644
--- a/arch/riscv/include/asm/csr.h
+++ b/arch/riscv/include/asm/csr.h
@@ -24,6 +24,12 @@
 #define SR_FS_CLEAN	_AC(0x00004000, UL)
 #define SR_FS_DIRTY	_AC(0x00006000, UL)
 
+#define SR_VS           _AC(0x00000600, UL) /* Vector Status */
+#define SR_VS_OFF       _AC(0x00000000, UL)
+#define SR_VS_INITIAL   _AC(0x00000200, UL)
+#define SR_VS_CLEAN     _AC(0x00000400, UL)
+#define SR_VS_DIRTY     _AC(0x00000600, UL)
+
 #define SR_XS		_AC(0x00018000, UL) /* Extension Status */
 #define SR_XS_OFF	_AC(0x00000000, UL)
 #define SR_XS_INITIAL	_AC(0x00008000, UL)
@@ -31,9 +37,9 @@
 #define SR_XS_DIRTY	_AC(0x00018000, UL)
 
 #ifndef CONFIG_64BIT
-#define SR_SD		_AC(0x80000000, UL) /* FS/XS dirty */
+#define SR_SD		_AC(0x80000000, UL) /* FS/VS/XS dirty */
 #else
-#define SR_SD		_AC(0x8000000000000000, UL) /* FS/XS dirty */
+#define SR_SD		_AC(0x8000000000000000, UL) /* FS/VS/XS dirty */
 #endif
 
 /* SATP flags */
@@ -120,6 +126,12 @@
 #define CSR_MIMPID		0xf13
 #define CSR_MHARTID		0xf14
 
+#define CSR_VSTART		0x8
+#define CSR_VCSR		0xf
+#define CSR_VL			0xc20
+#define CSR_VTYPE		0xc21
+#define CSR_VLENB		0xc22
+
 #ifdef CONFIG_RISCV_M_MODE
 # define CSR_STATUS	CSR_MSTATUS
 # define CSR_IE		CSR_MIE
-- 
2.31.1


